scripts/Muon/GUI/Common/plot_widget/plot_widget_presenter.py

```diff
@@ -83,7 +94,73 @@ def handle_data_updated(self):
             keys = self._model.create_tiled_keys(tiled_by)
             self._figure_presenter.create_tiled_plot(keys, tiled_by)
 
-        self.plot_all_selected_data(autoscale=False, hold_on=False)
+        self.plot_all_selected_data(autoscale=autoscale, hold_on=False)
+
+    def update_plot(self, autoscale=False):
+        if self.context.gui_context['PlotMode'] == PlotMode.Data:
+            self.handle_data_updated(autoscale=autoscale)
+        elif self.context.gui_context['PlotMode'] == PlotMode.Fitting:  # Plot the displayed workspace
+            self.handle_plot_selected_fits(
+                self._get_selected_fit_workspaces(), autoscale
+            )
+
+    def handle_plot_mode_changed(self, plot_mode : PlotMode):
+        if isinstance(self.context, FrequencyDomainAnalysisContext):
```

Comment:
I have seen this a few times. Would it be worth making it a method of the context?

Reply:
I don't think you gain much by adding a method to the context identifying what type of context it is when python as a language allows you to easily check the type of objects. The ideally we should be redeucing the number of these checks going forward as well as we differentiate between the different GUI's by more clearly seperating there code into different classes.